src/lib/annotations/drawing/DrawingModeController.js

```diff
@@ -0,0 +1,303 @@
+     *
+     * @inheritdoc
+     * @public
+     * @param {AnnotationThread} thread - The thread to register with the controller
+     * @return {void}
+     */
+    registerThread(thread) {
+        if (!thread || !thread.location) {
+            return;
+        }
+
+        this.threads.insert(thread);
+    }
+
+    /**
+     * Unregister a previously registered thread that has been assigned a location
+     *
+     * @inheritdoc
+     * @public
+     * @param {AnnotationThread} thread - The thread to unregister with the controller
+     * @return {void}
+     */
+    unregisterThread(thread) {
+        if (!thread || !thread.location) {
+            return;
+        }
+
+        this.threads.remove(thread);
+    }
+
+    /**
+     * Binds custom event listeners for a thread.
+     *
+     * @inheritdoc
+     * @protected
+     * @param {AnnotationThread} thread - Thread to bind events to
+     * @return {void}
+     */
+    bindCustomListenersOnThread(thread) {
+        if (!thread) {
+            return;
+        }
+
+        super.bindCustomListenersOnThread(thread);
+
+        // On save, add the thread to the Rbush, on delete, remove it from the Rbush
+        thread.addListener('annotationsaved', () => this.registerThread(thread));
+        thread.addListener('threaddeleted', () => this.unregisterThread(thread));
+    }
+
+    /**
+     * Set up and return the necessary handlers for the annotation mode
+     *
+     * @inheritdoc
+     * @protected
+     * @return {Array} An array where each element is an object containing the object that will emit the event,
+     *                 the type of events to listen for, and the callback
+     */
+    setupAndGetHandlers() {
+        const handlers = [];
+        /* eslint-disable require-jsdoc */
+        const locationFunction = (event) => this.annotator.getLocationFromEvent(event, TYPES.point);
+        /* eslint-enable require-jsdoc */
+
+        // Setup
+        this.currentThread = this.annotator.createAnnotationThread([], {}, TYPES.draw);
+        this.bindCustomListenersOnThread(this.currentThread);
+
+        // Get handlers
+        handlers.push(
+            {
+                type: ['mousemove', 'touchmove'],
+                func: annotatorUtil.eventToLocationHandler(locationFunction, this.currentThread.handleMove),
+                eventObj: this.annotator.annotatedElement
+            },
+            {
+                type: ['mousedown', 'touchstart'],
+                func: annotatorUtil.eventToLocationHandler(locationFunction, this.currentThread.handleStart),
+                eventObj: this.annotator.annotatedElement
+            },
+            {
+                type: ['mouseup', 'touchcancel', 'touchend'],
+                func: annotatorUtil.eventToLocationHandler(locationFunction, this.currentThread.handleStop),
+                eventObj: this.annotator.annotatedElement
+            }
+        );
+
+        if (this.postButtonEl) {
```

Comment:
move these into their own helper method and just call it for each button. May even make sense to move it to the base class!
```
addButtonHandlers(buttonEl, handler) {
  if (buttonEl) {
    const buttonHandler = {
      type: 'click,
      func: handler, 
      eventObj: buttonEl
    };
    handlers.push(buttonHandler);
}
```